### PR TITLE
Remove most py2 compat code

### PR DESCRIPTION
**Pull request by matthewturk**
This removes most of the python2 compatibility code.  There are a
handful of things that we no longer need to do, specifically lots of
`class Something(object):` invocations.

Unit tests pass; I anticipate there may be issues in answer testing
related to mistakes in importing or decode/encode of strings, and I look
forward to fixing those.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
One thing that I'm a little nervous about on the yt-4.0 branch, especially for a big invasive change like this, is that we don't have great test coverage for the frontends because travis and appveyor don't run tests with real data.

It would be great if we could get Jenkins running against the yt-4.0 branch next week.

**Comment by MatthewTurk** [comment]
OK, so this is a non-starter until then?

**Comment by ngoldbaum** [comment]
Make sure `python_requires` in `setup.py` gets updated, it can now just be `>=3.5` I think.

**Comment by ngoldbaum** [comment]
Getting this working would definitely be worthwhile since you'll have to do this annoying mechanical work anyway, there just might be more brokenness in the frontends or in code that only has tests that uses real data that you won't notice until jenkins gets turned on so this probably can't be merged until then.

**Comment by MatthewTurk** [comment]
To close the loop, this is dependent on #2208 and #2172 , and then (pending those tests passing) I believe it is ready to go.

**Comment by munkm** [comment]
I think this is now also dependent on yt-project/yt#2260

**Comment by jzuhone** [comment]
LGTM!

File: setup.py [hunk 1]
```diff
@@ -349,10 +349,10 @@ def run(self):
                  "Operating System :: POSIX :: Linux",
                  "Programming Language :: C",
                  "Programming Language :: Python :: 2",
```
Comment (chummels) [inline setup.py]: Don't we want to get rid of this line too? No Py2 coverage?

**Comment by matthewturk** [comment]
OK!  I have merged, and I think this is ready-for-review again.

**Pull request merged**